pants

```diff
@@ -404,6 +408,7 @@ function run_bootstrap_tools {
         "os_name=$(uname -s)"
         "arch=$(uname -m)"
         "python_path=${python}"
```

Comment:
Should we omit python_path from the key now?

Reply:
Or, now I'm wondering if realpath is actually a mistake - possibly we *do* want different virtualenvs that symlink to the same interpreter to have different cache keys?

Reply:
I think the realpath is required to ensure this key works to reliably stop issues like https://github.com/pantsbuild/actions/issues/5. That is: the cached directory includes a symlink to the system, so the key should change if the literal target of the symlink changes, and since `virtualenv` seems to flatten symlinks (rather than having symlinks to symlinks), the key should too.

As you say, maybe different virtualenvs that happen to link to the same interpreter should end up with different keys, and I think `python_path` may capture that. That said, I do also wonder if `python_path` is no longer necessary. 🤔

Reply:
I guess even the same venv could change over time, if someone gets really fiddly. There is no easy way to truly fingerprint a python interpreter's full state. But yeah, let's leave python_path in as an extra bulwark.